### PR TITLE
Atomic/discovery.py: Use login credentials if available to obtain token

### PR DESCRIPTION
**Pull request by baude**
In the case of an Atomic registry, we need to obtain a token by using
basic auth.  The login credentials are the same as what is used for
docker login.

### CODE REVIEW

**Comment by baude** [comment]
This should address https://github.com/projectatomic/atomic/issues/815

File: Atomic/backends/_docker.py [hunk 1]
```diff
@@ -97,8 +97,11 @@ def _make_image(self, image, img_struct, deep=False, remote=False):
         img_obj = Image(image, remote=remote)
         img_obj.backend = self
 
+        if img_struct is None:
+            return img_obj
+
         if not remote:
-            img_obj.id = img_struct['Id']
+            img_obj.id = img_struct['Id'] or None
```
Comment (rhatdan) [inline Atomic/backends/_docker.py]: Explain this syntax?  Would img_struct['Id'] have to be None to get the second option?

File: Atomic/discovery.py [hunk 2]
```diff
@@ -139,7 +157,7 @@ def get_auth_headers(self):
             self._create_self_auth_headers()
         return self.auth_headers
 
-    def debug(self, msg):
+    def write_debug(self, msg):
```
Comment (rhatdan) [inline Atomic/discovery.py]: Why are you changing this function name?

**Comment by rhatdan** [comment]
@aweiteka PTAL

File: Atomic/discovery.py (hunk 2 above)
Comment (baude) [inline Atomic/discovery.py]: Because self.debug is a bool and it was causing some confusion.

File: Atomic/discovery.py [hunk 3]
```diff
@@ -111,24 +114,39 @@ def load_local_tokens():
...
+        else:
+            results = requests.get(url, headers=headers, verify=self.verify)
+        self.write_debug("GET_RESPONSE_REASON: {}".format(results.reason))
+        if results._content: # pylint: disable=protected-access
+            self.write_debug("GET_RESPONSE_CONTENT: {}".format(str(results._content))) # pylint: disable=protected-access
         if results.reason == "Unauthorized" and self.needs_auth:
             # Auth registries dont tell us if credentials are bad or the GET URL is bad
             raise RegistryAuthError("Unable to authenticate to {} or repository, image, "
                                     "or tags names are bad".format(self.registry))
+        self.write_debug("")
```
Comment (aweiteka) [inline Atomic/discovery.py]: Is this intended?

File: Atomic/backends/_docker.py (hunk 1 above)
Comment (baude) [inline Atomic/backends/_docker.py]: This is a leftover debug.  Removed.

File: Atomic/discovery.py (hunk 3 above)
Comment (baude) [inline Atomic/discovery.py]: yeah, it is a linebreak when multiple requests are done and --debug is used...else it all flows together.

**Comment by rhatdan** [comment]
Still broken for me


```
./atomic --debug install --system dwalsh/etcdNamespace(_class=<class 'Atomic.install.Install'>, args=[], assumeyes=False, debug=True, display=False, func='install', image='dwalsh/etcd', name=None, opt1=None, opt2=None, opt3=None, remote=None, setvalues=None, system=True)
ReturnTuple(return_code=1, stdout='', stderr='time="2017-01-19T14:56:28-05:00" level=fatal msg="unauthorized: authentication required" \n')
Traceback (most recent call last):
  File "./atomic", line 188, in <module>
    sys.exit(_func())
  File "/home/dwalsh/atomic/Atomic/install.py", line 97, in install
    return self.syscontainers.install(self.image, self.name)
  File "/home/dwalsh/atomic/Atomic/syscontainers.py", line 178, in install
    image = self._pull_image_to_ostree(repo, image, False)
  File "/home/dwalsh/atomic/Atomic/syscontainers.py", line 146, in _pull_image_to_ostree
    self._check_system_oci_image(repo, image, upgrade)
  File "/home/dwalsh/atomic/Atomic/syscontainers.py", line 1159, in _check_system_oci_image
    manifest = self._skopeo_get_manifest(img)
  File "/home/dwalsh/atomic/Atomic/syscontainers.py", line 1001, in _skopeo_get_manifest
    return util.skopeo_inspect(img, args)
  File "/home/dwalsh/atomic/Atomic/util.py", line 267, in skopeo_inspect
    raise ValueError(results)
ValueError: ReturnTuple(return_code=1, stdout='', stderr='time="2017-01-19T14:56:28-05:00" level=fatal msg="unauthorized: authentication required" \n')
```

**Comment by baude** [comment]
Closing

**Pull request closed without merging**